### PR TITLE
rgw: optimize function parse_raw_oid

### PR DESCRIPTION
**Pull request by qrGitHub**
drop redundant checks

### CODE REVIEW

File: src/rgw/rgw_common.h [hunk 1]
```diff
@@ -1647,11 +1647,11 @@ struct rgw_obj_key {
       return true;
     }
 
-    if (oid[0] != '_' || oid.size() < 3) // for namespace, min size would be 3: _x_
```
Comment (qrGitHub) [inline src/rgw/rgw_common.h]: repeat with line 1640

File: src/rgw/rgw_common.h [hunk 2]
```diff
@@ -1647,11 +1647,11 @@ struct rgw_obj_key {
...
+    if (oid.size() < 3) // for namespace, min size would be 3: _x_
       return false;
 
-    int pos = oid.find('_', 1);
-    if (pos <= 1) // if it starts with __, it's not in our namespace
```
Comment (qrGitHub) [inline src/rgw/rgw_common.h]: oid[1] != '_'，so search at index 2.

**Comment by qrGitHub** [comment]
Hi @cbodley, would you help to review this? Thanks.

**Comment by qrGitHub** [comment]
Hi @cbodley, oid is a string, why we need to  check for empty in this function?

**Comment by cbodley** [comment]
> Hi @cbodley, oid is a string, why we need to check for empty in this function?

okay thanks, according to [string::operator[]](http://en.cppreference.com/w/cpp/string/basic_string/operator_at), `str[str.size()]` will give you a valid reference to the null character. so you're right, accessing `oid[0]` is safe here without a check

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-01-09-1802](https://trello.com/c/VlBAyUMv/225-wip-yuri3-testing-2018-01-09-1802)

**Pull request merged**